### PR TITLE
Migrate `TestBrokerTrigger` and `TestBrokerWithConfig` broker e2e test to new rekt test

### PR DESCRIPTION
**Pull request by creydr**
This is the first of a series of PRs to migrate the e2e tests in `test/e2e_broker` to the new reconciler-test framework.

## Proposed Changes
- :broom: Migrate `TestBrokerWithConfig` test

### CODE REVIEW

**Comment by creydr** [comment]
```
Running go test with args: -count=1 -race -tags=e2e -timeout=1h ./test/e2e_new_broker/...
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Setup/Create_broker_config (0.13s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Setup/Install_broker (0.15s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Broker_ready (3.13s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Topic_is_ready (0.06s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Replication_factor (0.04s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Number_of_partitions (0.04s)
PASS test/e2e_new_broker.TestBrokerWithConfig (5.80s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Setup/Create_broker_config (0.14s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Setup/Install_broker (0.18s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Setup/Install_sink (25.67s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Setup/Create_trigger (0.15s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Requirement/Send_matching_event (3.57s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Requirement/Send_non-matching_event (3.56s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Assert/[Stable/MUST]Broker_ready (0.04s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Assert/[Stable/MUST]Trigger_ready (0.04s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Assert/[Stable/MUST]Matching_event_received (0.00s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Assert/[Stable/MUST]Non_matching_event_is_not_received (20.00s)
PASS test/e2e_new_broker.TestBrokerTrigger (55.41s)
PASS test/e2e_new_broker

DONE 18 tests in 58.421s
Finished run, return code is 0
```

**Comment by creydr** [comment]
From [reconciler-tests_eventing-kafka-broker_main/1620400568868540416](https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka-broker/2931/reconciler-tests_eventing-kafka-broker_main/1620400568868540416):

```
Running go test with args: -count=1 -race -tags=e2e -timeout=1h ./test/e2e_new_broker/...
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Setup/Create_broker_config (0.14s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Setup/Install_broker (0.19s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Setup/Install_sink (23.89s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Setup/Create_trigger (0.16s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Requirement/Send_event (4.44s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Broker_ready (0.05s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Trigger_ready (0.05s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Event_received (4.00s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Topic_is_ready (0.07s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Replication_factor (0.04s)
PASS test/e2e_new_broker.TestBrokerWithConfig/Trigger_filters_events/Assert/[Stable/MUST]Number_of_partitions (0.04s)
PASS test/e2e_new_broker.TestBrokerWithConfig (35.15s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Setup/Create_broker_config (0.14s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Setup/Install_broker (0.18s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Setup/Install_sink (23.93s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Setup/Create_trigger (0.16s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Requirement/Send_matching_event (4.85s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Requirement/Send_non-matching_event (3.59s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Assert/[Stable/MUST]Broker_ready (0.04s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Assert/[Stable/MUST]Trigger_ready (0.04s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Assert/[Stable/MUST]Matching_event_received (0.00s)
PASS test/e2e_new_broker.TestBrokerTrigger/Trigger_filters_events/Assert/[Stable/MUST]Non_matching_event_is_not_received (20.00s)
PASS test/e2e_new_broker.TestBrokerTrigger (55.16s)
PASS test/e2e_new_broker

DONE 23 tests in 58.356s
```

File: test/reconciler-tests.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ if [ "${BROKER_CLASS}" == "KafkaNamespaced" ]; then
 fi
 
 go_test_e2e -timeout=1h ./test/e2e_new/... || fail_test "E2E (new) suite failed"
-
+go_test_e2e -tags=e2e -timeout=1h ./test/e2e_new_broker/... || fail_test "E2E (new - Broker) suite failed"
```
Comment (pierDipi) [inline test/reconciler-tests.sh:45]: Can't we use `e2e_new` package?

Comment (creydr) [reply test/reconciler-tests.sh]: updated the PR

File: test/e2e_broker/broker_trigger_test.go [hunk 2]
```diff
@@ -1,273 +0,0 @@
...
-	testingpkg "knative.dev/eventing-kafka-broker/test/pkg"
-	brokertest "knative.dev/eventing-kafka-broker/test/pkg/broker"
-	kafkatest "knative.dev/eventing-kafka-broker/test/pkg/kafka"
-	eventing "knative.dev/eventing/pkg/apis/eventing/v1"
-	testlib "knative.dev/eventing/test/lib"
-	"knative.dev/eventing/test/lib/recordevents"
-	"knative.dev/eventing/test/lib/resources"
-)
-
-func TestBrokerTrigger(t *testing.T) {
```
Comment (creydr) [inline test/e2e_broker/broker_trigger_test.go:44]: Didn't migrate this test, as this seems to test only default values and the trigger test should be covered by the conformance tests. The other test (`TestBrokerWithConfig`), tests the same with custom values

File: test/rekt/resources/kafkatopic/types.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type topic struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec topicSpec `json:"spec,omitempty"`
+}
+type topicSpec struct {
```
Comment (pierDipi) [inline test/rekt/resources/kafkatopic/types.go:29]:
```suggestion
}

type topicSpec struct {
```

File: test/rekt/resources/kafkatopic/topic.go [hunk 4]
```diff
@@ -69,6 +75,84 @@ func IsReady(name string, timings ...time.Duration) feature.StepFn {
...
+				return false, fmt.Errorf("could not convert unstructured to topic: %w", err)
+			}
+
+			if topicObj.Spec.Partitions != numPartitions {
+				t.Logf("Kafkatopic %s/%s does not have the expected number of partitions of %d. Has %d instead.", name, kafkaNamespace, topicObj.Spec.Partitions, numPartitions)
+				// keep polling
+				return false, nil
+			} else {
+				return true, nil
+			}
```
Comment (pierDipi) [inline test/rekt/resources/kafkatopic/topic.go:147]:
```suggestion
			if topicObj.Spec.Partitions != numPartitions {
				t.Logf("Kafkatopic %s/%s does not have the expected number of partitions of %d. Has %d instead.", name, kafkaNamespace, topicObj.Spec.Partitions, numPartitions)
				// keep polling
				return false, nil
			}
			return true, nil
```

**Comment by pierDipi** [comment]
2 nits

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**